### PR TITLE
tile resource support and refactoring

### PR DESCRIPTION
**Pull request by uniemimu**
Code cleanups and support for tile resource and related telemetry
originated node labels.

Co-authored-by: Tuomas Katila <tuomas.katila@intel.com>
Co-authored-by: Ukri Niemimuukko <ukri.niemimuukko@intel.com>

### CODE REVIEW

**Review by togashidm — COMMENTED** [review]
Great work! very minor things. 
In general, it looks good to me.

File: gpu-aware-scheduling/pkg/gpuscheduler/node_resource_cache.go [hunk 1]
```diff
@@ -281,35 +296,106 @@ func (c *Cache) checkPodResourceAdjustment(containerRequests []resourceMap,
...
+			return err
+		}
+
+		for _, cardName := range cardNames {
+			_, ok := nodeRes[cardName]
+			if !ok {
+				nodeRes[cardName] = resourceMap{}
+			}
+
+			if adj { // add
```
[the remark this answers is not in the document]
Comment (uniemimu) [reply gpu-aware-scheduling/pkg/gpuscheduler/node_resource_cache.go:317]: fixed

File: gpu-aware-scheduling/README.md [hunk 2]
```diff
@@ -123,6 +128,12 @@ GAS Scheduler Extender is set up to use in-Cluster config in order to access the
 Additionally GAS Scheduler Extender listens on a TLS endpoint which requires a cert and a key to be supplied.
 These are passed to the executable using command line flags. In the provided deployment these certs are added in a Kubernetes secret which is mounted in the pod and passed as flags to the executable from there.
 
+## License
+
+[Apache License, Version 2.0](./LICENSE). All of the source code required to build the GPU Aware Scheduling is available under Open Source
+licenses. The source code files identify external Go modules used. The binary is distributed as a container image on DockerHub. The
```
[the remark this answers is not in the document]
Comment (uniemimu) [reply gpu-aware-scheduling/README.md]: added

File: gpu-aware-scheduling/README.md [hunk 3]
```diff
@@ -37,7 +38,7 @@ You should follow extender configuration instructions from the
 use GPU Aware Scheduling configurations, which can be found in the [deploy/extender-configuration](deploy/extender-configuration) folder.
 
 #### Deploy GAS
-GPU Aware Scheduling uses go modules. It requires Go 1.16 with modules enabled in order to build. GAS has been tested with Kubernetes 1.22.
+GPU Aware Scheduling uses go modules. It requires Go 1.17 with modules enabled in order to build. GAS has been tested with Kubernetes 1.22.
 A yaml file for GAS is contained in the deploy folder along with its service and RBAC roles and permissions.
 
 **Note:** If run without the unsafe flag a secret called extender-secret will need to be created with the cert and key for the TLS endpoint.
```
[the remark this answers is not in the document]
Comment (uniemimu) [reply gpu-aware-scheduling/README.md]: indeed, removed the remark about unsafe

**Review by madalazar — APPROVED** [review]
LGTM!

**Review by togashidm — APPROVED** [review]

**Pull request merged**